### ISSUE TITLE
Add new gas and return data API to Computation object.

### ISSUE THREAD

**Issue by pipermerriam** [ethereum/py-evm#164]
### What is wrong?

With the introduction of the `REVERT` opcode in the byzantium fork rules we can no longer assume that the presence of `computation.error` means that gas has been fully depleted or that there is no return data.

This has resulted in the following lines of code being repeated in various places.

* https://github.com/pipermerriam/py-evm/blob/piper/byzantium/evm/vm/computation.py#L247
* https://github.com/pipermerriam/py-evm/blob/piper/byzantium/evm/vm/computation.py#L256
* https://github.com/pipermerriam/py-evm/blob/piper/byzantium/evm/logic/call.py#L114
* https://github.com/pipermerriam/py-evm/blob/piper/byzantium/evm/logic/call.py#L122
* https://github.com/pipermerriam/py-evm/blob/piper/byzantium/evm/vm/computation.py#L165

Some of this logic repetition would be better served by a dedicated API on the `Computation` object.

### How can it be fixed

Add the following two new APIs to the `Computation` object.

```python
def.should_burn_gas():
    return self.error and self.error.burns_gas

def should_erase_return_data():
    return self.error and self.error.zeros_return_data
```

*if* you would also like to cleanup the slighly confusing name `zeros_return_data` and change it to `erases_return_data` go for it.

**Comment by carver** [comment]
I had to read this a few places before I realized `zero` was a verb here. Maybe `should_erase_return_data`?

**Comment by frankcash** [comment]
The reference lines in the description are broken

**Comment by pipermerriam** [comment]
fixed in #181
